### PR TITLE
Add pyeda

### PR DESCRIPTION
**Pull request by lorenzleutgeb**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by lorenzleutgeb** [comment]
@conda-forge/help-python-c kindly asking for your review. Thanks.

**Comment by lorenzleutgeb** [comment]
I'd like to get this package working on Apple's M1 computers. I tried to follow https://conda-forge.org/blog/2020/10/29/macos-arm64/#how-to-add-a-osx-arm64-build-to-a-feedstock but I can't find any ARM binaries in the build artifacts. This is my first contribution to conda-forge, please help me out.

**Comment by zklaus** [comment]
Thanks for working on this, @lorenzleutgeb! Note that the article you link is about
> How to add a osx-arm64 build **to a feedstock**

in other words, you need a feedstock first. In conda-forge, packages are never added to the channel directly from staged-recipes, but only from their respective feedstocks, which will be created for pyeda when this PR is merged.

The CI jobs here are basically just checks for functioning builds to ensure that we have a good starting point in the feedstock. I think it is impossible to get an arm64 build in this (or any staged-recipes) PR. Focus on getting the three supported build here shaped up and then add the arm64 one in the feedstock once that is created.

**Comment by lorenzleutgeb** [comment]
@zklaus thanks, so if I understand correctly then, this PR first needs to get merged.

> Focus on getting the three supported build here shaped up and then add the arm64 one in the feedstock once that is created.

From what I understand looking at Azure output, all three builds seem to be fine. Is there anything left for me to do to get this merged?

File: recipes/pyeda/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  script_env:
+    - CFLAGS=-Wno-incompatible-pointer-types
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ stdlib("c") }}
+    - clang
```
Comment (zklaus) [inline recipes/pyeda/meta.yaml:22]: Why do you need `clang` here?

Comment (lorenzleutgeb) [reply recipes/pyeda/meta.yaml:22]: AFAIK it's required for compiling PicoSAT and Espresso. I'll try without.

**Review by zklaus — APPROVED** [review]
LGTM. I don't have merge rights though, help from @conda-forge/help-python-c is needed.

**Comment by ocefpaf** [comment]
@zklaus you should have merge rights here. Let me know if you want to be added to the staged-recipes team and we can fix that

**Comment by lorenzleutgeb** [comment]
@ocefpaf thanks a lot!

**Pull request merged**